src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAICrafting.java

```diff
@@ -194,6 +194,24 @@ protected IAIState getNextCraftingState()
         return GET_RECIPE;
     }
 
+    private boolean hasTooManyExternalItemsInInv(final IRecipeStorage currentRecipeStorage)
```

Comment:
could we please move this and the function below into a static item/inventory util for crafters? They dont need be part of the instance of the crafting AI. Also javadoc^^